### PR TITLE
Product Subscriptions: Remove legacy subscription row and screen code

### LINKED ISSUE #11246: Remove `.subscription` type row and code related to it
Remove the legacy subscription row and the detail screen

| Product form | Detail |
|--------|--------|
| Simulator Screenshot - iPhone 14 Pro - 2023-11-22 at 13 17 06 | Simulator Screenshot - iPhone 14 Pro - 2023-11-22 at 13 17 09 |

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #11246

## Description

We don't show the legacy subscription row in the product form anymore. It is being replaced with new rows such as Free trial and Expiry date. 

This PR removed the unused subscription row related code. 

## Testing instructions
- Smoke test that the subscription products feature works as before. Internal test plan - pe5sF9-2ew-p2
- CI should pass. 

## Screenshots
NA

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by hafizrahman** [comment]
Smoke test against test plan results in this one issue found:

https://github.com/woocommerce/woocommerce-ios/issues/11258

**Comment by selanthiraiyan** [comment]
> Smoke test against test plan results in this one issue found:
> https://github.com/woocommerce/woocommerce-ios/issues/11258

Thanks for testing, @hafizrahman!

Is this issue happening only in this PR? (i.e. After removing the legacy code)

File: WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactory+ProductCreationTests.swift [hunk 1]
```diff
@@ -7,7 +7,7 @@ import Fakes
 final class ProductFormActionsFactory_ProductCreationTests: XCTestCase {
     func test_product_type_is_editable_for_core_product_types() {
         // Product types supported in core.
-        let coreProductTypes: [ProductType] = [.simple, .variable, .affiliate, .grouped]
+        let coreProductTypes: [ProductType] = [.simple, .variable, .affiliate, .grouped, .subscription, .variableSubscription]
```
Comment (hafizrahman) [inline WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactory+ProductCreationTests.swift]: Here technically subscription/variable subscription is not a core product type, instead it's a type supported in Woo Extensions. However the test is to confirm that it is editable in the app, so indeed they might belong here.

wdyt about splitting it into its own variable? 

```
let nonCoreButEditableProductTypes: [ProductType] = [.subscription, .variableSubscription]
```

Or maybe include it in a separate unit test? e.g:

```
func test_product_type_is_editable_for_non_core_product_types() {
...
}
```

**Review by hafizrahman — COMMENTED** [review]
Nice work finding all the old `.subscription` code and removing them!

I left a comment for your attention, also made a new issue found from smoke testing. The issue is unrelated to this PR, though, so probably better handled in a separate PR.

**Comment by hafizrahman** [comment]
> Is this issue happening only in this PR? (i.e. After removing the legacy code)

I tested it only in this branch. I see you've merged trunk, let me re-test that.

File: WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactory+ProductCreationTests.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactory+ProductCreationTests.swift]: Nice observation, Hafiz! I moved the subscription product types out of this core unit test. https://github.com/woocommerce/woocommerce-ios/pull/11248/commits/ed0229bf377e1a30a76a08b0fe96ada480afa9f3

**Comment by hafizrahman** [comment]
@selanthiraiyan sadly the conversion issue is still there after re-testing.

**Review by hafizrahman — APPROVED** [review]
This should be good to go (assuming we plan to tackle the found issue separately)

**Comment by selanthiraiyan** [comment]
Thanks for the review, @hafizrahman! 

Considering that https://github.com/woocommerce/woocommerce-ios/issues/11258  is not introduced by the changes from this PR, let us merge these changes and address the issue separately.

**Pull request merged**